api/src/pcapi/core/providers/titelive_apply_new_filters_on_history.py

```diff
@@ -0,0 +1,348 @@
+                gtl_id = row[COLUMN_INDICES["CODE_GTL"]]
+                code_support = row[COLUMN_INDICES["CODE_SUPPORT"]]
+                id_lectorat = row[COLUMN_INDICES["IDLECTORAT"]]
+                taux_tva = row[COLUMN_INDICES["TAUX_TVA"]]
+                taux_tva = taux_tva.replace(".", ",")  #  fix decimal separator different from titelive_things files.
+                title = row[COLUMN_INDICES["TITLE"]]
+
+                ineligibility_reason = get_ineligibility_reason(
+                    gtl_id=gtl_id,
+                    code_support=code_support,
+                    id_lectorat=id_lectorat,
+                    taux_tva=taux_tva,
+                    title=title,
+                    comment=comments_dict[ean] if ean in comments_dict else "",
+                )
+
+                if ineligibility_reason:
+                    if ean in product_whitelist_eans:
+                        whitelisted_eans.append(ean)
+                    else:
+                        reasons[ineligibility_reason] = (
+                            {} if ineligibility_reason not in reasons else reasons[ineligibility_reason]
+                        ) | {
+                            "eans": []
+                            if ineligibility_reason not in reasons
+                            else reasons[ineligibility_reason]["eans"] + [ean],
+                            "total": 0
+                            if ineligibility_reason not in reasons
+                            else reasons[ineligibility_reason]["total"] + 1,
+                        }
+                        rejected_eans.append(ean)
+                        emails += reject_inappropriate_products(ean, send_booking_cancellation_emails=False, dry=dry)
+
+        logger.info(
+            "Total EANs rejected: %s | %s",
+            len(rejected_eans),
+            rejected_eans,
+        )
+        logger.info(
+            "Total EANs whitelisted in our database: %s | %s", len(product_whitelist_eans), product_whitelist_eans
+        )
+        logger.info("Total EANs whitelisted found in CSV: %s | %s", len(whitelisted_eans), whitelisted_eans)
+        logger.info("Details: %s", reasons)
+
+        with open(CANCELLED_BOOKING_EMAILS_FILE_PATH, "wb+") as cancelled_bookings_emails_csv:
+            emails = list(set(emails))
+            logger.info("Total email with bookings: %s", len(emails))
+            cancelled_bookings_emails_csv.write("\n".join(emails).encode())
+            cancelled_bookings_emails_csv.close()
+            logger.info("File with email written: %s", CANCELLED_BOOKING_EMAILS_FILE_PATH)
+
+        if dry:
+            logger.info("[dryrun] Rerun without --dry-run to apply those changes into the database")
+
+
+if __name__ == "__main__":
+    with app.app_context():
+        start = time.time()
+        dry_run = False
+
+        if len(sys.argv) > 1:
+            if sys.argv[1] in ["-d", "--dry-run"]:
+                dry_run = True
```

Comment:
J'en avais certainement parlé dans un autre script : je suis un couard, je préfère quand le "dry run" est par défaut : 

```suggestion
        dry_run = True
        if "--save" in sys.argv[1:]:
            dry_run = False
```

Cette modification ne coûte pas cher et ça évite une boulette particulièrement bête.

Reply:
Finalement, cette proposition faisait crash, et plutôt que chercher a corriger, je suis directement parti sur cette proposition de @Meewan : https://github.com/pass-culture/pass-culture-main/pull/7410#discussion_r1281996694